yarn-project/foundation/src/serialize/free_funcs.ts

```diff
@@ -147,6 +147,34 @@ export function from2Fields(field1: Fr, field2: Fr): Buffer {
   return Buffer.concat([originalPart1, originalPart2]);
 }
 
+/**
+ * Stores 248 bits of information in 1 field.
+ * @param buf - 32 or 31 bytes of data
+ * @returns 1 field element
+ */
+export function toTruncField(buf: Buffer): [Fr] {
+  if (buf.length !== 32 && buf.length !== 31) {
+    throw new Error('Buffer must be 31 or 32 bytes');
+  }
+
+  return [Fr.fromBuffer(buf.subarray(0, 31))];
+}
+
+/**
+ * Reconstructs the original 31 bytes of data from 1 truncated field element.
+ * @param field - field element
+ * @returns 31 bytes of data as a Buffer
+ */
+export function fromTruncField(field: Fr): Buffer {
+  // the below is adapted from toBufferBE - can import here if appropriate
+  const hex = field.toBigInt().toString(16);
+  const buffer = Buffer.from(hex.padStart(62, '0').slice(0, 62), 'hex');
+  if (buffer.length > 31) {
```

Comment:
Shouldn't this always have length 31 if it's a buffer from string.slice(0, 62)?
Maybe it can be `field.toBuffer()`. Throw if `buffer[0] != 0`?

Reply:
I thought that as well - it's copied from the logic here (`foundation/src/bigint-buffer/index`):

```ts
/**
 * Convert a BigInt to a big-endian buffer.
 * @param num - The BigInt to convert.
 * @param width - The number of bytes that the resulting buffer should be.
 * @returns A big-endian buffer representation of num.
 */
export function toBufferBE(num: bigint, width: number): Buffer {
  if (num < BigInt(0)) {
    throw new Error(`Cannot convert negative bigint ${num.toString()} to buffer with toBufferBE.`);
  }
  const hex = num.toString(16);
  const buffer = Buffer.from(hex.padStart(width * 2, '0').slice(0, width * 2), 'hex');
  if (buffer.length > width) {
    throw new Error(`Number ${num.toString(16)} does not fit in ${width}`);
  }
  return buffer;
}
```
so I assumed the width check must do something!
Either way I think `buffer[0]` is cleaner, and then truncate to 31 bytes, so I will make the change.